core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java

```diff
@@ -91,27 +68,44 @@ public void close() {
         try {
             ((CloseableHttpClient)getHttpClient()).close();
         } catch (IOException e) {
-            InternalLogger.INSTANCE.error("Failed to close http client, exception: %s", e.toString());
+            InternalLogger.INSTANCE.error("Failed to close http httpClient, exception: %s", e.toString());
+        } finally {
+            synchronized (lock) {
+                if (connectionManager != null) {
+                    connectionManager.shutdown();
+                }
+            }
         }
     }
 
     @Override
     public HttpClient getHttpClient() {
-        if (!isClientInitialized) {
-            synchronized (this) {
-                if (!isClientInitialized) {
-                    while (httpClientRef.get() == null) {
-                        try {
-                            TimeUnit.MILLISECONDS.sleep(3);
-                        } catch (InterruptedException e){
-                        }
-                    }
-                    isClientInitialized = true;
+        CloseableHttpClient result = httpClient;
+        if (result == null) {
+            synchronized (lock) {
+                connectionManager = initializeConnectionManager();
+                result = httpClient;
```

Comment:
Why do we need this line here? In the flow I didn't see that httpClient / result variable is updated anywhere. This seems redundant.

Reply:
It's a standard pattern. A good explanation here: https://stackoverflow.com/questions/3578604/how-to-solve-the-double-checked-locking-is-broken-declaration-in-java

Reply:
Thanks makes sense